nucleo_f429zi: multiple networking tests failing

**Describe the bug**
It looks as though a recent change in the STM32 Ethernet HAL has broken a number of networking tests for `nucleo_f429zi`.

Please also mention any information which could help others to understand
the problem you're facing:
 - **What target platform are you using?**
 Twister in CI
 - **What have you tried to diagnose or workaround this issue?**
 - looked at twister output in CI

**To Reproduce**
Steps to reproduce the behavior:
1. `twister -i -p nucleo_f429zi -s samples/net/civetweb/http_server/sample.net.civetweb.http_server`
2. `twister -i -p nucleo_f429zi -s tests/net/socket/websocket/net.socket.websocket`
3. `twister -i -p nucleo_f429zi -s tests/net/socket/udp/net.socket.udp`
4. `twister -i -p nucleo_f429zi -s tests/net/socket/udp/net.socket.udp.preempt`
5. `twister -i -p nucleo_f429zi -s tests/net/socket/tls/net.socket.tls.preempt`
7. `twister -i -p nucleo_f429zi -s tests/net/socket/tls/net.socket.tls`
8. `twister -i -p nucleo_f429zi -s tests/net/socket/tcp/net.socket.tcp.preempt`
9. `twister -i -p nucleo_f429zi -s tests/net/socket/tcp/net.socket.tcp`
10. `twister -i -p nucleo_f429zi -s tests/net/socket/socketpair/net.socket.socketpair`
11. `twister -i -p nucleo_f429zi -s tests/net/socket/socketpair/net.socket.socketpair.newlib`
12. `twister -i -p nucleo_f429zi -s tests/net/socket/select/net.socket.select.preempt`
13. `twister -i -p nucleo_f429zi -s tests/net/socket/select/net.socket.select`
14. `twister -i -p nucleo_f429zi -s tests/net/socket/register/net.socket.register.tls`
15. `twister -i -p nucleo_f429zi -s tests/net/socket/register/net.socket.register`
16. `twister -i -p nucleo_f429zi -s tests/net/socket/poll/net.socket.poll`
17. `twister -i -p nucleo_f429zi -s tests/net/socket/misc/net.socket.misc.userspace`
18. `twister -i -p nucleo_f429zi -s tests/net/socket/misc/net.socket.misc`
19. `twister -i -p nucleo_f429zi -s tests/net/socket/af_packet_ipproto_raw/net.socket.af_packet.ipproto_raw`
20. `twister -i -p nucleo_f429zi -s tests/net/socket/af_packet/net.socket.af_packet`
21. `twister -i -p nucleo_f429zi -s samples/net/sockets/websocket_client/sample.net.sockets.websocket_client`
22. `twister -i -p nucleo_f429zi -s samples/net/sockets/socketpair/sample.net.sockets.socketpair`
...
See errors

There are 35 errors in total, and they are all on `nucleo_f429zi`.

```
- 583 of 668 test configurations passed (94.34%), 35 failed, 50 skipped with 0 warnings in 3190.26 seconds
```

**Expected behavior**
This should build without error.

**Impact**
What impact does this issue have on your progress (e.g., annoyance, showstopper)

**Logs and console output**
```
FAILED: zephyr/drivers/ethernet/CMakeFiles/drivers__ethernet.dir/eth_stm32_hal.c.obj 
ccache /opt/toolchains/zephyr-sdk-0.14.0/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DCORE_CM4 -DHSE_VALUE=8000000 -DKERNEL -DSTM32F429xx -DUSE_FULL_LL_DRIVER -DUSE_HAL_DRIVER -D_FORTIFY_SOURCE=2 -D__PROGRAM_START -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I../../../../../../../samples/net/civetweb/http_server/../common/include -I../../../../../../../include -Izephyr/include/generated -I../../../../../../../soc/arm/st_stm32/stm32f4 -I../../../../../../../include/posix -I../../../../../../../drivers -I../../../../../../../soc/arm/st_stm32/common -I../../../../../../../subsys/net/l2 -I../../../../../../../subsys/net/lib/sockets/. -I../../../../../../../subsys/net/lib/dns/. -I/__w/zephyr/modules/lib/civetweb/include -I/__w/zephyr/modules/hal/cmsis/CMSIS/Core/Include -I/__w/zephyr/modules/hal/atmel/include -I/__w/zephyr/modules/hal/stm32/stm32cube/stm32f4xx/soc -I/__w/zephyr/modules/hal/stm32/stm32cube/stm32f4xx/drivers/include -I/__w/zephyr/modules/hal/stm32/stm32cube/stm32f4xx/drivers/include/Legacy -I/__w/zephyr/modules/hal/stm32/stm32cube/common_ll/include -isystem ../../../../../../../lib/libc/minimal/include -isystem /opt/toolchains/zephyr-sdk-0.14.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/include -isystem /opt/toolchains/zephyr-sdk-0.14.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/include-fixed -Os -imacros /__w/zephyr/zephyr/twister-out/nucleo_f429zi/samples/net/civetweb/http_server/sample.net.civetweb.http_server/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m4 -mthumb -mabi=aapcs -mfp16-format=ieee -imacros /__w/zephyr/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -Werror -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-reorder-functions -fno-defer-pop -fmacro-prefix-map=/__w/zephyr/zephyr/samples/net/civetweb/http_server=CMAKE_SOURCE_DIR -fmacro-prefix-map=/__w/zephyr/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/__w/zephyr=WEST_TOPDIR -ffunction-sections -fdata-sections -std=c99 -nostdinc -include /__w/zephyr/zephyr/samples/net/civetweb/http_server/../common/include/libc_extensions.h -MD -MT zephyr/drivers/ethernet/CMakeFiles/drivers__ethernet.dir/eth_stm32_hal.c.obj -MF zephyr/drivers/ethernet/CMakeFiles/drivers__ethernet.dir/eth_stm32_hal.c.obj.d -o zephyr/drivers/ethernet/CMakeFiles/drivers__ethernet.dir/eth_stm32_hal.c.obj -c /__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c: In function 'read_eth_phy_register':
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c:105:47: error: passing argument 3 of 'HAL_ETH_ReadPHYRegister' makes integer from pointer without a cast [-Werror=int-conversion]
  105 |  return HAL_ETH_ReadPHYRegister(heth, PHYReg, RegVal);
      |                                               ^~~~~~
      |                                               |
      |                                               uint32_t * {aka unsigned int *}
In file included from /__w/zephyr/modules/hal/stm32/stm32cube/stm32f4xx/drivers/include/stm32f4xx_hal_conf.h:332,
                 from /__w/zephyr/modules/hal/stm32/stm32cube/stm32f4xx/drivers/include/stm32f4xx_hal.h:29,
                 from /__w/zephyr/modules/hal/stm32/stm32cube/stm32f4xx/soc/stm32f4xx.h:287,
                 from ../../../../../../../soc/arm/st_stm32/stm32f4/soc.h:23,
                 from ../../../../../../../include/arch/arm/aarch32/cortex_m/cmsis.h:17,
                 from ../../../../../../../include/arch/arm/aarch32/mpu/arm_mpu_v7m.h:11,
                 from ../../../../../../../include/arch/arm/aarch32/mpu/arm_mpu.h:14,
                 from ../../../../../../../include/arch/arm/aarch32/arch.h:194,
                 from ../../../../../../../include/arch/cpu.h:19,
                 from ../../../../../../../include/kernel_includes.h:33,
                 from ../../../../../../../include/kernel.h:17,
                 from ../../../../../../../include/zephyr.h:18,
                 from /__w/zephyr/zephyr/samples/net/civetweb/http_server/../common/include/libc_extensions.h:12,
                 from <command-line>:
/__w/zephyr/modules/hal/stm32/stm32cube/stm32f4xx/drivers/include/stm32f4xx_hal_eth.h:2067:95: note: expected 'uint32_t' {aka 'unsigned int'} but argument is of type 'uint32_t *' {aka 'unsigned int *'}
 2067 | HAL_StatusTypeDef HAL_ETH_ReadPHYRegister(ETH_HandleTypeDef *heth, uint32_t PHYAddr, uint32_t PHYReg,
      |                                                                                      ~~~~~~~~~^~~~~~
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c:105:9: error: too few arguments to function 'HAL_ETH_ReadPHYRegister'
  105 |  return HAL_ETH_ReadPHYRegister(heth, PHYReg, RegVal);
      |         ^~~~~~~~~~~~~~~~~~~~~~~
In file included from /__w/zephyr/modules/hal/stm32/stm32cube/stm32f4xx/drivers/include/stm32f4xx_hal_conf.h:332,
                 from /__w/zephyr/modules/hal/stm32/stm32cube/stm32f4xx/drivers/include/stm32f4xx_hal.h:29,
                 from /__w/zephyr/modules/hal/stm32/stm32cube/stm32f4xx/soc/stm32f4xx.h:287,
                 from ../../../../../../../soc/arm/st_stm32/stm32f4/soc.h:23,
                 from ../../../../../../../include/arch/arm/aarch32/cortex_m/cmsis.h:17,
                 from ../../../../../../../include/arch/arm/aarch32/mpu/arm_mpu_v7m.h:11,
                 from ../../../../../../../include/arch/arm/aarch32/mpu/arm_mpu.h:14,
                 from ../../../../../../../include/arch/arm/aarch32/arch.h:194,
                 from ../../../../../../../include/arch/cpu.h:19,
                 from ../../../../../../../include/kernel_includes.h:33,
                 from ../../../../../../../include/kernel.h:17,
                 from ../../../../../../../include/zephyr.h:18,
                 from /__w/zephyr/zephyr/samples/net/civetweb/http_server/../common/include/libc_extensions.h:12,
                 from <command-line>:
/__w/zephyr/modules/hal/stm32/stm32cube/stm32f4xx/drivers/include/stm32f4xx_hal_eth.h:2067:19: note: declared here
 2067 | HAL_StatusTypeDef HAL_ETH_ReadPHYRegister(ETH_HandleTypeDef *heth, uint32_t PHYAddr, uint32_t PHYReg,
      |                   ^~~~~~~~~~~~~~~~~~~~~~~
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c: In function 'eth_tx':
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c:209:22: error: 'ETH_HandleTypeDef' has no member named 'TxDesc'; did you mean 'TxDescList'?
  209 |  dma_tx_desc = heth->TxDesc;
      |                      ^~~~~~
      |                      TxDescList
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c:66:55: error: 'ETH_DMADescTypeDef' has no member named 'Status'
   66 | #define IS_ETH_DMATXDESC_OWN(dma_tx_desc) (dma_tx_desc->Status & \
      |                                                       ^~
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c:66:55: note: in definition of macro 'IS_ETH_DMATXDESC_OWN'
   66 | #define IS_ETH_DMATXDESC_OWN(dma_tx_desc) (dma_tx_desc->Status & \
      |                                                       ^~
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c:231:38: error: 'ETH_DMADescTypeDef' has no member named 'Buffer1Addr'
  231 |  dma_buffer = (uint8_t *)(dma_tx_desc->Buffer1Addr);
      |                                      ^~
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c:307:12: error: implicit declaration of function 'HAL_ETH_TransmitFrame'; did you mean 'HAL_ETH_Transmit_IT'? [-Werror=implicit-function-declaration]
  307 |  hal_ret = HAL_ETH_TransmitFrame(heth, total_len);
      |            ^~~~~~~~~~~~~~~~~~~~~
      |            HAL_ETH_Transmit_IT
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c: In function 'eth_rx':
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c:468:12: error: implicit declaration of function 'HAL_ETH_GetReceivedFrame_IT' [-Werror=implicit-function-declaration]
  468 |  hal_ret = HAL_ETH_GetReceivedFrame_IT(heth);
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c:474:18: error: 'ETH_HandleTypeDef' has no member named 'RxFrameInfos'
  474 |  total_len = heth->RxFrameInfos.length;
      |                  ^~
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c:475:30: error: 'ETH_HandleTypeDef' has no member named 'RxFrameInfos'
  475 |  dma_buffer = (uint8_t *)heth->RxFrameInfos.buffer;
      |                              ^~
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c:544:20: error: 'ETH_HandleTypeDef' has no member named 'RxFrameInfos'
  544 |  dma_rx_desc = heth->RxFrameInfos.FSRxDesc;
      |                    ^~
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c:546:26: error: 'ETH_HandleTypeDef' has no member named 'RxFrameInfos'
  546 |  for (int i = 0; i < heth->RxFrameInfos.SegCount; i++) {
      |                          ^~
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c:547:14: error: 'ETH_DMADescTypeDef' has no member named 'Status'
  547 |   dma_rx_desc->Status |= ETH_DMARXDESC_OWN;
      |              ^~
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c:549:16: error: 'ETH_DMADescTypeDef' has no member named 'Buffer2NextDescAddr'
  549 |    (dma_rx_desc->Buffer2NextDescAddr);
      |                ^~
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c:553:6: error: 'ETH_HandleTypeDef' has no member named 'RxFrameInfos'
  553 |  heth->RxFrameInfos.SegCount = 0;
      |      ^~
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c: In function 'eth_initialize':
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c:888:2: error: implicit declaration of function 'HAL_ETH_DMATxDescListInit' [-Werror=implicit-function-declaration]
  888 |  HAL_ETH_DMATxDescListInit(heth, dma_tx_desc_tab,
      |  ^~~~~~~~~~~~~~~~~~~~~~~~~
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c:890:2: error: implicit declaration of function 'HAL_ETH_DMARxDescListInit' [-Werror=implicit-function-declaration]
  890 |  HAL_ETH_DMARxDescListInit(heth, dma_rx_desc_tab,
      |  ^~~~~~~~~~~~~~~~~~~~~~~~~
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c: At top level:
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c:1080:5: error: 'ETH_InitTypeDef' has no member named 'AutoNegotiation'
 1080 |    .AutoNegotiation = ETH_AUTONEGOTIATION_ENABLE,
      |     ^~~~~~~~~~~~~~~
In file included from /__w/zephyr/modules/hal/stm32/stm32cube/stm32f4xx/drivers/include/stm32f4xx_hal_conf.h:332,
                 from /__w/zephyr/modules/hal/stm32/stm32cube/stm32f4xx/drivers/include/stm32f4xx_hal.h:29,
                 from /__w/zephyr/modules/hal/stm32/stm32cube/stm32f4xx/soc/stm32f4xx.h:287,
                 from ../../../../../../../soc/arm/st_stm32/stm32f4/soc.h:23,
                 from ../../../../../../../include/arch/arm/aarch32/cortex_m/cmsis.h:17,
                 from ../../../../../../../include/arch/arm/aarch32/mpu/arm_mpu_v7m.h:11,
                 from ../../../../../../../include/arch/arm/aarch32/mpu/arm_mpu.h:14,
                 from ../../../../../../../include/arch/arm/aarch32/arch.h:194,
                 from ../../../../../../../include/arch/cpu.h:19,
                 from ../../../../../../../include/kernel_includes.h:33,
                 from ../../../../../../../include/kernel.h:17,
                 from ../../../../../../../include/zephyr.h:18,
                 from /__w/zephyr/zephyr/samples/net/civetweb/http_server/../common/include/libc_extensions.h:12,
                 from <command-line>:
/__w/zephyr/modules/hal/stm32/stm32cube/stm32f4xx/drivers/include/stm32f4xx_hal_eth.h:1319:40: error: initialization of 'uint8_t *' {aka 'unsigned char *'} from 'unsigned int' makes pointer from integer without a cast [-Werror=int-conversion]
 1319 | #define ETH_AUTONEGOTIATION_ENABLE     0x00000001U
      |                                        ^~~~~~~~~~~
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c:1080:23: note: in expansion of macro 'ETH_AUTONEGOTIATION_ENABLE'
 1080 |    .AutoNegotiation = ETH_AUTONEGOTIATION_ENABLE,
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~
/__w/zephyr/modules/hal/stm32/stm32cube/stm32f4xx/drivers/include/stm32f4xx_hal_eth.h:1319:40: note: (near initialization for 'eth0_data.heth.Init.MACAddr')
 1319 | #define ETH_AUTONEGOTIATION_ENABLE     0x00000001U
      |                                        ^~~~~~~~~~~
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c:1080:23: note: in expansion of macro 'ETH_AUTONEGOTIATION_ENABLE'
 1080 |    .AutoNegotiation = ETH_AUTONEGOTIATION_ENABLE,
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c:1094:5: error: 'ETH_InitTypeDef' has no member named 'PhyAddress'
 1094 |    .PhyAddress = PHY_ADDR,
      |     ^~~~~~~~~~
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c:1095:5: error: 'ETH_InitTypeDef' has no member named 'RxMode'
 1095 |    .RxMode = ETH_RXINTERRUPT_MODE,
      |     ^~~~~~
In file included from /__w/zephyr/modules/hal/stm32/stm32cube/stm32f4xx/drivers/include/stm32f4xx_hal_conf.h:332,
                 from /__w/zephyr/modules/hal/stm32/stm32cube/stm32f4xx/drivers/include/stm32f4xx_hal.h:29,
                 from /__w/zephyr/modules/hal/stm32/stm32cube/stm32f4xx/soc/stm32f4xx.h:287,
                 from ../../../../../../../soc/arm/st_stm32/stm32f4/soc.h:23,
                 from ../../../../../../../include/arch/arm/aarch32/cortex_m/cmsis.h:17,
                 from ../../../../../../../include/arch/arm/aarch32/mpu/arm_mpu_v7m.h:11,
                 from ../../../../../../../include/arch/arm/aarch32/mpu/arm_mpu.h:14,
                 from ../../../../../../../include/arch/arm/aarch32/arch.h:194,
                 from ../../../../../../../include/arch/cpu.h:19,
                 from ../../../../../../../include/kernel_includes.h:33,
                 from ../../../../../../../include/kernel.h:17,
                 from ../../../../../../../include/zephyr.h:18,
                 from /__w/zephyr/zephyr/samples/net/civetweb/http_server/../common/include/libc_extensions.h:12,
                 from <command-line>:
/__w/zephyr/modules/hal/stm32/stm32cube/stm32f4xx/drivers/include/stm32f4xx_hal_eth.h:1329:33: error: initialization of 'ETH_DMADescTypeDef *' from 'unsigned int' makes pointer from integer without a cast [-Werror=int-conversion]
 1329 | #define ETH_RXINTERRUPT_MODE    0x00000001U
      |                                 ^~~~~~~~~~~
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c:1095:14: note: in expansion of macro 'ETH_RXINTERRUPT_MODE'
 1095 |    .RxMode = ETH_RXINTERRUPT_MODE,
      |              ^~~~~~~~~~~~~~~~~~~~
/__w/zephyr/modules/hal/stm32/stm32cube/stm32f4xx/drivers/include/stm32f4xx_hal_eth.h:1329:33: note: (near initialization for 'eth0_data.heth.Init.TxDesc')
 1329 | #define ETH_RXINTERRUPT_MODE    0x00000001U
      |                                 ^~~~~~~~~~~
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c:1095:14: note: in expansion of macro 'ETH_RXINTERRUPT_MODE'
 1095 |    .RxMode = ETH_RXINTERRUPT_MODE,
      |              ^~~~~~~~~~~~~~~~~~~~
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c:1096:5: error: 'ETH_InitTypeDef' has no member named 'ChecksumMode'
 1096 |    .ChecksumMode = ETH_CHECKSUM_BY_SOFTWARE,
      |     ^~~~~~~~~~~~
In file included from /__w/zephyr/modules/hal/stm32/stm32cube/stm32f4xx/drivers/include/stm32f4xx_hal_conf.h:332,
                 from /__w/zephyr/modules/hal/stm32/stm32cube/stm32f4xx/drivers/include/stm32f4xx_hal.h:29,
                 from /__w/zephyr/modules/hal/stm32/stm32cube/stm32f4xx/soc/stm32f4xx.h:287,
                 from ../../../../../../../soc/arm/st_stm32/stm32f4/soc.h:23,
                 from ../../../../../../../include/arch/arm/aarch32/cortex_m/cmsis.h:17,
                 from ../../../../../../../include/arch/arm/aarch32/mpu/arm_mpu_v7m.h:11,
                 from ../../../../../../../include/arch/arm/aarch32/mpu/arm_mpu.h:14,
                 from ../../../../../../../include/arch/arm/aarch32/arch.h:194,
                 from ../../../../../../../include/arch/cpu.h:19,
                 from ../../../../../../../include/kernel_includes.h:33,
                 from ../../../../../../../include/kernel.h:17,
                 from ../../../../../../../include/zephyr.h:18,
                 from /__w/zephyr/zephyr/samples/net/civetweb/http_server/../common/include/libc_extensions.h:12,
                 from <command-line>:
/__w/zephyr/modules/hal/stm32/stm32cube/stm32f4xx/drivers/include/stm32f4xx_hal_eth.h:1338:39: error: initialization of 'ETH_DMADescTypeDef *' from 'unsigned int' makes pointer from integer without a cast [-Werror=int-conversion]
 1338 | #define ETH_CHECKSUM_BY_SOFTWARE      0x00000001U
      |                                       ^~~~~~~~~~~
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c:1096:20: note: in expansion of macro 'ETH_CHECKSUM_BY_SOFTWARE'
 1096 |    .ChecksumMode = ETH_CHECKSUM_BY_SOFTWARE,
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~
/__w/zephyr/modules/hal/stm32/stm32cube/stm32f4xx/drivers/include/stm32f4xx_hal_eth.h:1338:39: note: (near initialization for 'eth0_data.heth.Init.RxDesc')
 1338 | #define ETH_CHECKSUM_BY_SOFTWARE      0x00000001U
      |                                       ^~~~~~~~~~~
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c:1096:20: note: in expansion of macro 'ETH_CHECKSUM_BY_SOFTWARE'
 1096 |    .ChecksumMode = ETH_CHECKSUM_BY_SOFTWARE,
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c: In function 'read_eth_phy_register':
/__w/zephyr/zephyr/drivers/ethernet/eth_stm32_hal.c:107:1: error: control reaches end of non-void function [-Werror=return-type]
  107 | }
      | ^
cc1: all warnings being treated as errors
```

**Environment (please complete the following information):**
 - OS: Linux / macOS / CI
 - Toolchain: Zephyr SDK v0.14.0
 - Commit SHA or Version used: 018f59e46fcfb1cf90d9ba44735ebc00590cef2e

**Additional context**

--- a/west.yml
+++ b/west.yml
@@ -129,7 +129,7 @@ manifest:
       groups:
         - hal
     - name: hal_stm32
-      revision: 13964e0af173f19fafad3692869e8486f3c2043d
+      revision: 5945d50882a5133e6cab918d1214c3be373b2c6a
       path: modules/hal/stm32
       groups:
         - hal

